tests/components/environment_canada/test_weather.py

```diff
@@ -0,0 +1,126 @@
+"""Test weather."""
+
+from datetime import UTC, datetime
+import json
+from unittest.mock import MagicMock, patch
+
+from homeassistant.components.environment_canada.weather import get_forecast
+
+from tests.common import load_fixture
+
+
+def test_forecast_daily() -> None:
+    """Test basic forecast."""
+
+    ec_data = json.loads(
+        load_fixture("environment_canada/current_conditions_data.json")
+    )
+
+    # Remove the previous day's night forecast.
+    del ec_data["daily_forecasts"][0]
+
+    mock = MagicMock()
+    mock.daily_forecasts = ec_data["daily_forecasts"]
+
+    with patch(
```

Comment:
You should not mock internals, use freezer instead to move the time to the necessary dates.
Also this should be tested by setting up the integration, not by testing the respective methods themselves.

Take a look at some other weather integration tests to see how it's done there.

Reply:
Given this is a bug fix I suggest I drop the new tests and add them back properly after.

Reply:
Hard to know it's fixing the problem without a test...
It would seem you can use `init_integration` (move it to `init.py`) and use that to load the integration including this fixture.
Then it would be a quite easy test to get the states and the forecast from the service to see it does what it should.

Reply:
Thanks for the suggestion. Done!

Reply:
Could you help a bit? I'm stuck understanding freeze_time. I use `freeze_time` as shown on the first line. In the test right under the annotation I print `now()` and its the second line -- all good. But when I get into the code and get the time it has now changed to the third line.

The code works as desired. The test does not work as desired as I test for `hour < 6` which fails when running the test.

Feels like a timezone thing, but any combos I've tried to resolve this have not worked as desired.

```
@freeze_time("2022-10-04 01:00:00+00:00")
HAFakeDatetime(2022, 10, 4, 1, 0)
HAFakeDatetime(2022, 10, 3, 18, 0, tzinfo=zoneinfo.ZoneInfo(key='US/Pacific'))
```

I also tried `@freeze_time(datetime.datetime(2022, 10, 4, 1, tzinfo=datetime.UTC))` which yielded identical results.

I get Pacific is 7 hours off of UTC. I don't understand why the `freeze_time` specified as UTC is showing up as Pacific.

Reply:
You are using `dt_util.now()` which is providing the time with the timezone you're in instead of `dt_util.utcnow()`. 

You're probably fine running it locally (I guess your timezone is close to UTC) but in the tests it's using a completely different timezone (US) and therefore you get this strange thing.

Without looking to careful I guess your comparing time with a timezone (now()) with times without a timezone (incoming data).

Reply:
My timezone is EDT (Toronto/New York), so UTC-4. Ideally I would get the time of the weather station but there's no easy way to do that (that I've found yet).

The problem I'm trying to solve is that after midnight the weather forecast from the previous evening is still reported by Environment Canada and we want that report to show up on the correct day. They don't put timestamps in the forecasts.

So the check I have in the code under test is to check if the local time is between midnight and 6am.

Not suggesting I go back to patching the time, but when I was I'm pretty sure that the time didn't change timezones between the test and the CUT.

Last comment. I reading through issues in the freezegun repo. There's lots of TZ related issues but they seems to be mostly fixed -- they have not given a clue as to why this is happening though.

Reply:
I made a commit now to fix the tests.
But the `if now.hour < 6 and half_days[0]["temperature_class"] != "high"´ check is anything but robust.
That check depends very much on your HA timezone to be in proximity of the station you're interested in as otherwise that hour check might fall short.
We should try to fix that somehow.